### PR TITLE
[openstack-cloud-controller-manager] Delete the floating IP only when needed

### PR DESCRIPTION
**Pull request by lingxiankong**
**The binaries affected**:

IMPORTANT: Please also add the binary name in the title, e.g.
`[openstack-cloud-controller-manager]: Add UDP protocol support`
unless the PR affects multiple binaries.

- [x] openstack-cloud-controller-manager
- [ ] cinder-csi-plugin
- [ ] k8s-keystone-auth
- [ ] client-keystone-auth
- [ ] octavia-ingress-controller
- [ ] manila-csi-plugin
- [ ] manila-provisioner
- [ ] magnum-auto-healer
- [ ] barbican-kms-plugin

**What this PR does / why we need it**:
Delete the floating IP only if it was created dynamically by the controller manager.

The openstack-cloud-controller-manager now supports to specify an existing floating IP in `Spec.LoadBalancerIP` field, but that IP address is deleted by openstack-cloud-controller-manager when the Service is deleted which is not the expected behavior.

**Which issue this PR fixes**:
fixes #

**Special notes for reviewers**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go [hunk 1]
```diff
@@ -1839,9 +1840,17 @@ func (lbaas *LbaasV2) EnsureLoadBalancerDeleted(ctx context.Context, clusterName
...
-					return fmt.Errorf("failed to delete floating IP %s for loadbalancer VIP port %s: %v", fip.FloatingIP, portID, err)
+				matched, err := regexp.Match("Floating IP for Kubernetes external service", []byte(fip.Description))
+				if err != nil {
+					return err
+				}
+				if matched {
+					if err := floatingips.Delete(lbaas.network, fip.ID).ExtractErr(); err != nil {
+						return fmt.Errorf("failed to delete floating IP %s for loadbalancer VIP port %s: %v", fip.FloatingIP, portID, err)
+					}
 				}
```
Comment (jichenjc) [inline pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go]: should we add some log in !matched flag ?

Comment (lingxiankong) [reply pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go]: We could, but I didn't see it's very useful, it's obvious that only the floating IP whose description not satisfies the regex will not be touched. The condition answer is yes or no, the log is useful only if there are multiple conditions.

Comment (lingxiankong) [reply pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go]: Hi @jichenjc, are you ok if we merge this PR?

**Comment by ramineni** [comment]
cc @chrigl

**Comment by jichenjc** [comment]
yes, I think it's good to merge

**Pull request merged**